x/participationrewards/keeper/hooks.go

```diff
@@ -6,14 +6,27 @@ import (
 	epochstypes "github.com/ingenuity-build/quicksilver/x/epochs/types"
 )
 
+var epochsDeferred = int64(3)
+
 func (k Keeper) BeforeEpochStart(ctx sdk.Context, epochIdentifier string, epochNumber int64) {
 }
 
 func (k Keeper) AfterEpochEnd(ctx sdk.Context, epochIdentifier string, epochNumber int64) {
 	k.Logger(ctx).Info("distribute participation rewards...")
 
-	if epochNumber < 3 {
-		k.Logger(ctx).Info("defer...")
+	if epochNumber < epochsDeferred {
+		k.Logger(ctx).Info("defer...", "epoch", epochNumber)
+
+		// create snapshot of current intents for the next epoch boundary
+		// requires intents to be set, no intents no snapshot...
+		// further snapshots will be taken during
+		// ValidatorSelectionRewardsCallback;
+		for _, zone := range k.icsKeeper.AllZones(ctx) {
+			for _, di := range k.icsKeeper.AllOrdinalizedIntents(ctx, zone, false) {
```

Comment:
I was just looking at this.

Do we require snapshotted intents to be ordinalized? 

Presently, intents are store as normalized - but snapshotted are ordinalized - but I cannot see why this is. If this is intentional it should be documented; otherwise we should switch to `AllIntents()`.

Reply:
I think we started using ordinalized intents (sorted by weight) when we experienced some determinism issues in earlier testnets... safe to revert to AllIntents()?

Reply:
yes please - saves a whole bunch of KV lookups each epoch, and makes everything consistent.